publication-commons/src/main/java/no/unit/nva/publication/permissions/publication/grant/ClaimedChannelGrantStrategy.java

```diff
@@ -0,0 +1,50 @@
+package no.unit.nva.publication.permissions.publication.grant;
+
+import static no.unit.nva.publication.model.business.publicationchannel.ChannelPolicy.EVERYONE;
+import static no.unit.nva.publication.model.business.publicationchannel.ChannelPolicy.OWNER_ONLY;
+import java.net.URI;
+import no.unit.nva.model.PublicationOperation;
+import no.unit.nva.publication.model.business.Resource;
+import no.unit.nva.publication.model.business.UserInstance;
+import no.unit.nva.publication.model.business.publicationchannel.ChannelPolicy;
+import no.unit.nva.publication.permissions.publication.PublicationGrantStrategy;
+import no.unit.nva.publication.permissions.publication.PublicationStrategyBase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ClaimedChannelGrantStrategy extends PublicationStrategyBase implements PublicationGrantStrategy {
+
+    public static final Logger logger = LoggerFactory.getLogger(ClaimedChannelGrantStrategy.class);
+
+    public ClaimedChannelGrantStrategy(Resource resource, UserInstance userInstance) {
+        super(resource, userInstance);
+    }
+
+    @Override
+    public boolean allowsAction(PublicationOperation permission) {
+        var claimedPublicationChannel = resource.getPrioritizedClaimedPublicationChannel();
+        if (claimedPublicationChannel.isEmpty()) {
+            return false;
+        }
+
+        var publicationChannel = claimedPublicationChannel.get();
```

Comment:
Pull the following lines out, and combine the resulting method with the above as a ternary. The logic here is "if -> if -> if -> if", and we should name what we can to make this relatively complex chain explicit.